rails_app/app/interactors/send_newsletter.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+class SendNewsletter
+  include Interactor
+  include Delayed::RecurringJob
+
+  run_every 1.week
+  run_at "9:00am"
+  timezone "Europe/Lisbon"
+
+  def perform
+    self.call
+  end
+
+  def call
+    User.all.each do |user|
```

Comment:
In a big app this would use all the memory in a heartbeat. It's always better to parse potentially infinite collections in batches!

Reply:
https://api.rubyonrails.org/v7.0.4.2/classes/ActiveRecord/Batches.html#method-i-find_each

Reply:
👍